### PR TITLE
tikv: ensure safe-ts won't be max uint64

### LINKED ISSUE #1249: safe-ts is set to max uint64 after rolling restart tikv
After #793 , we try to get safe ts from pd http API. However, the semantics of the API returns `MaxUint64` when the safe ts is uninitalized, which is not same as previous `GetStoreSafeTS` RPC (of tikv). As a result, safe ts will be set to max uint64 after rolling restart tikv, which further leads to a large amount of stale read miss, because the calculated read ts will be now (the upper bound) in this case.

### PR DESCRIPTION
**Pull request by zyguan**
Try to fix #1249

with this PR, no stale read miss after rolling restart tikv:

without this fix:

### CODE REVIEW

**Review by cfzjywxk — APPROVED** [review]

File: tikv/kv_test.go [hunk 1]
```diff
@@ -105,22 +124,127 @@ func (c *storeSafeTsMockClient) CloseAddr(addr string) error {
...
+}
+
+func (c *mockPDHTTPClient) GetMinResolvedTSByStoresIDs(ctx context.Context, storeIDs []uint64) (uint64, map[uint64]uint64, error) {
+	if f := c.mockGetMinResolvedTSByStoresIDs.Load(); f != nil {
+		return (*f)(ctx, storeIDs)
+	}
+	return c.Client.GetMinResolvedTSByStoresIDs(ctx, storeIDs)
+}
+
+func until(limit int, interval time.Duration, predicate func() bool) bool {
```
Comment (you06) [inline tikv/kv_test.go:139]: Can we use `s.Eventually`?

File: tikv/kv_test.go [hunk 2]
```diff
@@ -105,22 +124,127 @@ func (c *storeSafeTsMockClient) CloseAddr(addr string) error {
...
+			} else {
+				m[id] = math.MaxUint64
+			}
+		}
+		return math.MaxUint64, m, nil
+	})
+	until(10, time.Second, func() bool {
+		ts := s.store.GetMinSafeTS("z1")
+		s.Require().False(math.MaxUint64 == ts)
+		return ts == uint64(10)
```
Comment (you06) [inline tikv/kv_test.go:244]: This `until` call will timeout and the predicate never returns true.

Comment (ekexium) [reply tikv/kv_test.go:244]: Should we check the return value of `until`?

Comment (zyguan) [reply tikv/kv_test.go:244]: Sorry, it should GetMinSafeTS from z2.

**Review by you06 — APPROVED** [review]

**Pull request merged**